azurerm/internal/services/sql/parse/sql_database.go

```diff
@@ -0,0 +1,41 @@
+package parse
+
+import (
+	"fmt"
+
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+)
+
+type SqlDatabaseId struct {
+	SubscriptionID string
+	ResourceGroup  string
+	ServerName     string
+	Name           string
+}
+
+func SqlDatabaseID(input string) (*SqlDatabaseId, error) {
+	id, err := azure.ParseAzureResourceID(input)
+	if err != nil {
+		return nil, fmt.Errorf("parsing Synapse Sql Pool ID %q: %+v", input, err)
+	}
+
+	sqlDatabaseId := SqlDatabaseId{
+		SubscriptionID: id.SubscriptionID,
+		ResourceGroup:  id.ResourceGroup,
+	}
+	if sqlDatabaseId.ServerName, err = id.PopSegment("servers"); err != nil {
+		return nil, err
+	}
+	if sqlDatabaseId.Name, err = id.PopSegment("databases"); err != nil {
+		return nil, err
+	}
+	if err := id.ValidateNoEmptySegments(input); err != nil {
+		return nil, err
+	}
+
+	return &sqlDatabaseId, nil
+}
+
+func (id *SqlDatabaseId) String() string {
+	return fmt.Sprintf("/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Sql/servers/%s/databases/%s", id.SubscriptionID, id.ResourceGroup, id.ServerName, id.Name)
+}
```

Comment:
we've got a ResourceID Formatter type which this can comply with now - [example](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/internal/services/frontdoor/parse/health_probe.go#L5-L22) and [example of the test asserting this conforms to the type](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/internal/services/frontdoor/parse/health_probe_test.go#L9) - can we update this?